sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/MonotonicallyIncreasingID.scala

```diff
@@ -81,3 +93,12 @@ case class MonotonicallyIncreasingID() extends LeafExpression with Nondeterminis
 
   override def sql: String = s"$prettyName()"
 }
+
+object MonotonicallyIncreasingID {
+  private def parseExpression(expr: Expression): Long = expr match {
+    case IntegerLiteral(i) => i.toLong
+    case NonNullLiteral(l, LongType) => l.toString.toLong
```

Comment:
```
object NonNullLiteral {
  def unapply(literal: Literal): Option[(Any, DataType)] = {
    Option(literal.value).map(_ => (literal.value, literal.dataType))
```
A tuple is returned by the extractor. I don't think l: Long would compile.

If the above doesn't match, I wonder why AnalysisException from the default case wasn't thrown.